### PR TITLE
feat: add Cosmwasm into experimental build

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

 - [x] Update SDK version to v0.45
 - [x] Add Cosmwasm into experimental build

Closes: #578 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -405,21 +425,83 @@ func NewRegenApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest
...
+			ibctransfertypes.ModuleName,
+		}, setCustomOrderBeginBlocker()...)...,
+	)
+	app.mm.SetOrderEndBlockers(
+		append([]string{
+			crisistypes.ModuleName,
+			govtypes.ModuleName,
+			stakingtypes.ModuleName,
+			ibchost.ModuleName,
+			ibctransfertypes.ModuleName,
```
Comment (ryanchristo) [inline app/app.go:457]: These are included twice here and below.

**Review by ryanchristo — COMMENTED** [review]
Looking good. A few comments and questions. Will follow up after manually testing.

File: app/stable_appconfig.go [hunk 2]
```diff
@@ -49,64 +44,39 @@ func setCustomKVStoreKeys() []string {
...
+func (app *RegenApp) setCustomAnteHandler(encCfg simappparams.EncodingConfig, wasmKey *sdk.KVStoreKey, _ *wasm.Config) (sdk.AnteHandler, error) {
+	return ante.NewAnteHandler(
+		ante.HandlerOptions{
+			AccountKeeper:   app.AccountKeeper,
+			BankKeeper:      app.BankKeeper,
+			SignModeHandler: encCfg.TxConfig.SignModeHandler(),
+			FeegrantKeeper:  app.FeeGrantKeeper,
+			SigGasConsumer:  ante.DefaultSigVerificationGasConsumer,
+		},
+	)
```
[the remark this answers is not in the document]
Comment (aleem1314) [reply app/stable_appconfig.go:70]: `IBCChannelKeeper`, `WasmConfig` and `TXCounterStoreKey` part of custom ante handler which is used in experimental config. For the stable config, we are using default ante [handler](https://github.com/regen-network/regen-ledger/pull/702/files#diff-0f1d2976054440336a576d47a44a37b80cdf6701dd9113012bce0e3c425819b7L464-L472).

**Review by ryanchristo — APPROVED** [review]
Looks good to me.

Also, manually tested cosmwasm contract support and was able to successfully store a binary, instantiate a contract, and execute messages using the [latest cw-plus contracts](https://github.com/CosmWasm/cw-plus/releases/tag/v0.11.1).

**Review by anilCSE — APPROVED** [review]
lgtm, we should also figure out a way to add custom wasm config to app.toml (simapp has an example)

**Review by anilCSE — APPROVED** [review]
lgtm, we should also figure out a way to add custom wasm config to app.toml (simapp has an example)

**Review by atheeshp — APPROVED** [review]
lgtm!

**Review by robert-zaremba — APPROVED** [review]
lgtm...

**Comment by ryanchristo** [comment]
Any ideas why this pull request is hanging on "Build Expected — Waiting for status to be reported"? I was hoping a new commit would fix the issue but it looks like it's still stuck.

**Comment by aleem1314** [comment]
> Any ideas why this pull request is hanging on "Build Expected — Waiting for status to be reported"? I was hoping a new commit would fix the issue but it looks like it's still stuck.

There is no github action with the `Build` name. I've updated it to use [matrix](https://github.com/regen-network/regen-ledger/pull/702/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R12-R29). Can we make `build-experimental` `build-stable`, `Tests / Cover report (stable-codecov)` and `Tests / Cover report (experimental-codecov)` CI jobs required?

**Comment by ryanchristo** [comment]
cc @clevinson can you update the required CI checks?

**Pull request merged**